lib/Graph/Context.cpp

```diff
@@ -70,46 +70,61 @@ Placeholder *Context::getPlaceholderByName(llvm::StringRef name) const {
   return nameIt->second;
 }
 
-void Context::insert(Placeholder *P, Tensor &&T) {
-  assert(!map_.count(P) && "Placeholder already registered");
+void Context::insert(Placeholder *P, std::unique_ptr<Tensor> T,
+                     bool overwrite) {
+  assert((overwrite || !map_.count(P)) && "Placeholder already registered");
+  map_[P] = std::move(T);
+  nameMap_[P->getName()] = P;
+}
+
+void Context::insert(Placeholder *P, Tensor &&T, bool overwrite) {
```

Comment:
Consider implementing this as

```
insert(P, llvm::make_unique<Tensor>(std::move(T)), overwrite);
```

if possible.

Reply:
Yeah I thought about this but it's like 3 lines so I don't know that it's worth it. Especially because I think we should remove this method in the future and just only manage tensors in contexts as unique_ptrs, no reason to have this then.